spec/jobs/dhis2/ethiopia_exporter_job_spec.rb

```diff
@@ -62,4 +90,81 @@
       end
     end
   end
+
+  describe "#segregate_and_format_treatment_data" do
+    it "returns the data segregated with treatment type" do
+      Timecop.freeze("April 25th 2024") do
+        create_list(:patient, 3, :hypertension, :under_care)
+        Reports::PatientState.refresh
+      end
+      patients = Reports::PatientState.all.to_a
+      output = described_class.new.send(:segregate_and_format_treatment_data, patients)
+      expect(output.count).to eq 2
+      expect(output["lsm"]).to eq patients.count
+      expect(output["pharma_management"]).to eq 0
+    end
+
+    it "returns 0 count with an empty data set" do
+      output = described_class.new.send(:segregate_and_format_treatment_data, [])
+      expect(output.count).to eq 2
+      expect(output["lsm"]).to eq 0
+      expect(output["pharma_management"]).to eq 0
+    end
+
+    it "also includes patients with lsm prescribed drugs" do
+      Timecop.freeze("April 25th 2024") do
+        create(:patient, :hypertension, :under_care)
+        patient_2 = create(:patient, :hypertension, :under_care)
+        create(:prescription_drug, patient: patient_2, recorded_at: Time.current - 1.month, name: "Lifestyle Management")
+        Reports::PatientVisit.refresh
+        Reports::PatientState.refresh
+      end
+      patients = Reports::PatientState.all.to_a
+      output = described_class.new.send(:segregate_and_format_treatment_data, patients)
+      expect(output.count).to eq 2
+      expect(output["lsm"]).to eq patients.count
+      expect(output["pharma_management"]).to eq 0
+    end
+  end
+
+  describe "#segregate_and_format_enrollment_data" do
+    it "returns the data segregated by enrollment time" do
+      Timecop.freeze("April 25th 2024") do
+        create_list(:patient, 3, :hypertension, :under_care)
+        Reports::PatientState.refresh
+      end
+      patients = Reports::PatientState.all.to_a
+      output = described_class.new.send(:segregate_and_format_enrollment_data, patients)
+      expect(output.count).to eq 2
+      expect(output["newly_enrolled"]).to eq 3
+      expect(output["previously_enrolled"]).to eq(patients.count - 3)
+    end
+
+    it "returns 0 count with an empty data set" do
+      output = described_class.new.send(:segregate_and_format_enrollment_data, [])
+      expect(output.count).to eq 2
+      expect(output["newly_enrolled"]).to eq 0
+      expect(output["previously_enrolled"]).to eq 0
+    end
+  end
+
+  describe "#segregate_and_format_cohort_data" do
+    it "returns the data segregated by enrollment time" do
+      Timecop.freeze("April 25th 2024") do
+        create(:patient, :hypertension, :controlled)
+        create(:patient, :hypertension, :uncontrolled)
+        create(:patient, :hypertension, :lost_to_follow_up)
+        create(:patient, :hypertension, :dead)
+        create(:patient, :hypertension, status: "migrated")
+        refresh_views
```

Comment:
[suggestion, nonblocking] Can we also replace this and other calls to refresh only the views that need to be refreshed? You have done it correctly for all the specs above this one, perhaps this was missed in the refactor?